pkg/process/metadata/workloadmeta/workloadmeta.go

```diff
@@ -0,0 +1,96 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package workloadmeta
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/languagedetection"
+	"github.com/DataDog/datadog-agent/pkg/languagedetection/languagemodels"
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+)
+
+// ProcessEntity is a placeholder workloadmeta.ProcessEntity.
+// It does not contain all the fields that the final entity will contain.
+type ProcessEntity struct {
+	pid      int32
+	cmdline  []string
+	language *languagemodels.Language
+}
+
+// WorkloadMetaExtractor handles enriching processes with
+type WorkloadMetaExtractor struct {
+	cache      map[string]*ProcessEntity
+	cacheMutex sync.RWMutex
+
+	grpcListener mockableGrpcListener
+}
+
+// NewWorkloadMetaExtractor constructs the WorkloadMetaExtractor.
+func NewWorkloadMetaExtractor() *WorkloadMetaExtractor {
+	return &WorkloadMetaExtractor{
+		cache:        make(map[string]*ProcessEntity),
+		grpcListener: newGrpcListener(),
+	}
+}
+
+// Extract detects the process language, creates a process entity, and sends that entity to WorkloadMeta
+func (w *WorkloadMetaExtractor) Extract(procs map[int32]*procutil.Process) {
+	procsToDetect := make([]*procutil.Process, 0, len(procs))
```

Comment:
What do you think of renaming `procsToDetect` to `newProcs` and the current `newProcs` to `newEntities`?

Reply:
Great suggestion! Updated